### ISSUE TITLE
photospline - look at potential clang false positive (Trac #987)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#981]
with clang-3.7

```text
[ 12%] Building C object photospline/CMakeFiles/photospline.dir/private/lib/bspline_multi.c.o
clang: warning: optimization flag '-fno-signaling-nans' is not supported
clang: warning: optimization flag '-fno-signaling-nans' is not supported
../../src/photospline/private/lib/bspline_multi.c:78:23: warning: variable 'sp' is uninitialized when used here [-Wuninitialized]
 if (__builtin_expect(sp & 15UL, 0))
                      ^~
../../src/photospline/private/lib/bspline_multi.c:77:36: note: initialize the variable 'sp' to silence this warning
 volatile register unsigned long sp __asm("esp");
                                   ^
                                    = 0
1 warning generated.
```

clang is possible getting confused by `__asm()`

Migrated from https://code.icecube.wisc.edu/ticket/987
```json
{
    "status": "closed", 
    "changetime": "2016-03-18T21:14:09", 
    "description": "with clang-3.7\n{{{\n[ 12%] Building C object photospline/CMakeFiles/photospline.dir/private/lib/bspline_multi.c.o\nclang: warning: optimization flag '-fno-signaling-nans' is not supported\nclang: warning: optimization flag '-fno-signaling-nans' is not supported\n../../src/photospline/private/lib/bspline_multi.c:78:23: warning: variable 'sp' is uninitialized when used here [-Wuninitialized]\n if (__builtin_expect(sp & 15UL, 0))\n                      ^~\n../../src/photospline/private/lib/bspline_multi.c:77:36: note: initialize the variable 'sp' to silence this warning\n volatile register unsigned long sp __asm(\"esp\");\n                                   ^\n                                    = 0\n1 warning generated.\n}}}\n\nclang is possible getting confused by `__asm()`", 
    "reporter": "nega", 
    "cc": "jvansanten", 
    "resolution": "fixed", 
    "_ts": "1458335649133028", 
    "component": "combo core", 
    "summary": "photospline - look at potential clang false positive", 
    "priority": "trivial", 
    "keywords": "", 
    "time": "2015-05-14T15:27:27", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150903T17:51:08: 
* nega changed priority from "normal" to "trivial"
* nega changed milestone from "" to "January 2016"

**Comment by icecube-trac** [comment]
Trac update at 20150903T17:52:58: 
* nega changed summary from "phys-services - look at potential clang false positive" to "photospline - look at potential clang false positive"
* nega changed cc from "" to "jvansanten"

**Comment by icecube-trac** [comment]
Trac update at 20150903T18:44:24: 
* jvansanten changed status from "new" to "closed"
* jvansanten commented:

> In changeset 137254:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="137254"
> Fix #981.
> 
> Clang did in fact leave this variable uninitialized without the second bit of
> extended inline asm. On a side note, the GCC/Linux bug this was intended to
> work around appears to be gone as of 4.4.7/RHEL6: in a million tries I always
> get a 16-byte-aligned stack.
> ```


* jvansanten changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20160318T21:14:09: 
* olivas commented:

> Milestone January 2016 deleted


* olivas changed milestone from "January 2016" to ""
